src/front/spv/mod.rs

```diff
@@ -2277,13 +2282,35 @@ impl<I: Iterator<Item = u32>> Parser<I> {
             _ => return Err(Error::UnsupportedType(lookup_type.handle)),
         };
         let class = map_storage_class(storage)?;
+        let is_storage = match module.types[ty].inner {
+            crate::TypeInner::Struct { .. } => class == crate::StorageClass::StorageBuffer,
+            crate::TypeInner::Image {
+                class: crate::ImageClass::Storage(_),
+                ..
+            } => true,
+            _ => false,
+        };
+
+        let storage_access = if is_storage {
+            let mut access = crate::StorageAccess::all();
```

Comment:
So if I understand correctly, the `StorageAccess` for Structs and Images are by default both Readable and Writable? I am more in favor of using it defensively and start with `StorageAccess::empty()`, or isn't this an expected behavior?

Reply:
this is SPIRV frontend: stuff is RW by default unless one of  those decorations is present
In the IR we are explicit about this with this PR :)